substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/InstrumentFeature.java

```diff
@@ -94,12 +97,32 @@ public void afterRegistration(AfterRegistrationAccess access) {
      * is absent. <br>
      * So this method looks for them in the same order.
      */
-    private void addPremainClass(String premainClass) {
+    private void addPremainClass(String javaagentOption) {
+        int separatorIndex = javaagentOption.indexOf("=");
+        String agent;
+        String premainClass = null;
+        String options = "";
+        // Get the agent file
+        if (separatorIndex == -1) {
+            agent = javaagentOption;
+        } else {
+            agent = javaagentOption.substring(0, separatorIndex);
+            options = javaagentOption.substring(separatorIndex + 1);
+        }
+        // Read MANIFEST in agent jar
+        try {
+            JarFile agentJarFile = new JarFile(agent);
+            premainClass = agentJarFile.getManifest().getMainAttributes().getValue("Premain-Class");
```

Comment:
What happens when there is no Manifest?

Reply:
At this moment, the java agent has been already loaded and run by JVM. 
If there is no Manifest, the JVM can't get started with `-javaagent`.
But in the proxy agent case, the actual agent will be loaded by -cp, we should check the existence of Manifest then.